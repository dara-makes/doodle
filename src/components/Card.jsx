import React from 'react';
import './Card.scss';

/**
 * Card - Displays each days' weather data.
 *
 * @param props
 * @returns {*}
 * @constructor
 */
const Card = props => {
  let day, temp;
  props.day ? ([day, temp] = [props.day, props.high]) : ([day, temp] = ['Now', props.temp]);

  return (
    <li className="card">
      <i className={`wi wi-yahoo-${props.code}`} />
      <h1 className="card-temp">
        {temp}Â°
        <span className="card-temp-low">
          {props.low}
        </span>
      </h1>
      <h2 className="card-day">
        {day}
      </h2>
      <p className="card-text">
        {props.text}
      </p>
    </li>
  );
};

export default Card;
