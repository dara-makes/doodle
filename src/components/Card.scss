@import "../app/mixin";

.card {
  display: grid;
  text-align: center;
  padding-top: 4em;
  opacity: 0.7;
  transition: all 300ms;

  .wi {
    font-size: 12vw;
  }

  &-temp {
    font-size: 8vw;
    font-weight: 700;
    margin-top: 1em;

    &-low {
      font-size: .75em;
      opacity: .5;
      margin-left: .25em;
    }
  }

  &-day {
    font-size: 7vw;
    font-weight: 100;
  }

  &-text {
    font-size: 5vw;
    font-weight: 100;
    margin-top: .15em;
  }

  &:first-child{
    .wi {
      font-size: 25vw;
    }
    .card-day {
      font-size: 12vw;
    }
  }

  @include breakpoint(small) {
    .wi {
      font-size: 2em;
    }

    &-temp {
      font-size: 1.5em;
      margin-top: 1.5em;
    }

    &-day {
      font-size: 1.25em;
    }

    &-text {
      font-size: 1em;
    }

    &:first-child {
      grid-column: span 2;
      .wi {
        font-size: 3em;
        transform: scale(2) translateY(-.25em);
      }
      .card-day {
        font-size: 2em;
      }
      .card-text {
        font-size: 1.25em;
      }
    }
  }

  @include breakpoint(medium) {
    &:first-child {
      grid-column: span 4;
    }
  }

  @include breakpoint(large) {
    &:first-child {
      grid-column: span 1;
    }
  }
}