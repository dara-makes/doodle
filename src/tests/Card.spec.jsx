import React from 'react';
import { shallow } from 'enzyme';
import Card from '../components/Card';

describe('Card', () => {
  it('should render card', () => {
    const cardComponent = shallow(<Card />);
    expect(cardComponent).toMatchSnapshot();
  });

  it('should contain weather data', () => {
    const cardComponent = shallow(<Card />);
    expect(cardComponent.find('i').length).toEqual(1);
    expect(cardComponent.find('.card-temp').length).toEqual(1);
    expect(cardComponent.find('.card-day').length).toEqual(1);
    expect(cardComponent.find('.card-text').length).toEqual(1);
  });
});
