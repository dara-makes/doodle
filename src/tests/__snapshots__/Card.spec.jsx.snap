// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render card 1`] = `
<li
  className="card"
>
  <i
    className="wi wi-yahoo-undefined"
  />
  <h1
    className="card-temp"
  >
    Â°
    <span
      className="card-temp-low"
    />
  </h1>
  <h2
    className="card-day"
  >
    Now
  </h2>
  <p
    className="card-text"
  />
</li>
`;
